src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,31 @@
+package duke.command;
+
+import duke.component.Storage;
+import duke.component.TaskList;
+import duke.component.Ui;
+import duke.exception.DukeException;
+
+/**
+ * Represents user commmands to chatbot.
+ * The 'Command' class supports operators (i) executing the command
+ * and (ii) checking if the bot has exited its conversation with the user.
+ */
+public abstract class Command {
+
+    /**
+     * Executes the command and print out respecive reponse.
+     *
+     * @param taskList List of the things user needs to do
+     * @param ui       Interface that interacts with the user
+     * @param storage  Stores the user input in a file
+     * @throws DukeException IOException if there is an error writing or reading file
+     */
+    public abstract String execute(TaskList taskList, Ui ui, Storage storage) throws DukeException;
+
+    /**
+     * Returns a boolean value on whether the program has exited.
+     *
+     * @return boolean of whether program has exited
+     */
+    public abstract boolean isExit();
+}
```

Comment:
Consider adding newlines at the end of your files, as it is a general convention. It doesn't really cause any issues beside making GitHub and checkstyle whine, but there is a reason behind doing so, https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline. Hope this gives some insight!